### PR TITLE
Add boost test lecture

### PR DESCRIPTION
**Pull request by uekerman**
## Description

Add slide and demo notes for Boost.Test lecture. I did not yet prepare notes for preCICE, but I am already at 90 minutes now.

## Checklist

- [x] I made sure that the markdown files are formatted properly.
- [x] I made sure that all hyperlinks are working.

- [x] I added the new content to the `README.md` inside the topics subdirectory, e.g., `01_version_control/README.md`.
- [x] I added the new content to the `timetable.md` in the root of this repository.
- [x] I make sure to update the content/links on the [homepage](https://github.com/Simulation-Software-Engineering/homepage).

### CODE REVIEW

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    - add `"${SRC_FILES}"` to test executable
+    - makes headers visible `target_include_directories(testsidemade PRIVATE ${PROJECT_SOURCE_DIR}/src)`
+    - link `Eigen3::Eigen` and also `yaml-cpp` (since used in `SRC_FILES`)
+
+- build ... problem that two "main" functions are defined
+    - Fix by:
+
+```cmake
+file(GLOB_RECURSE SRC_FILES CONFIGURE_DEPENDS src/*.cpp)
+list(REMOVE_ITEM SRC_FILES "${PROJECT_SOURCE_DIR}/src/main.cpp")
```
Comment (ajaust) [inline 05_testing_and_ci/boost_testing_sidemade_demo.md:104]: Maybe we will have a student that shows us an even better solution to remove `main.cpp` from the source files. :)

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+```
+
+- `cp ../config.yml testconfig.yml` and explain what test should do
+- expected data:
+
+```cpp
+const MatrixSolver::DecompositionType expectedDecompositionType{MatrixSolver::QR};
+const std::string                     expectedMatrixFileName{"../data/m3.csv"};
+const int                             expectedMatrixSize{3};
+```
```
Comment (ajaust) [inline 05_testing_and_ci/boost_testing_sidemade_demo.md:184]: Technically the code block should be indented with two spaces such that is assigned to the bullet point. That also applies to the other code blocks.
```suggestion
  ```cpp
  const MatrixSolver::DecompositionType expectedDecompositionType{MatrixSolver::QR};
  const std::string                     expectedMatrixFileName{"../data/m3.csv"};
  const int                             expectedMatrixSize{3};
  ```
```

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+BOOST_TEST(configuration.decompositionType == expectedDecompositionType);
+BOOST_TEST(configuration.matrixSize == expectedMatrixSize);
+```
+
+- adjust `CMakeLists.txt`
+    - add tests and explain filter
+
+```cmake
+add_test(NAME "MatrixSolverTests" COMMAND ${CMAKE_CURRENT_BINARY_DIR}/testsidemade --run_test=MatrixSolverTests/*)
+add_test(NAME "ConfigurationTests" COMMAND ${CMAKE_CURRENT_BINARY_DIR}/testsidemade --run_test=ConfigurationTests/*
```
Comment (ajaust) [inline 05_testing_and_ci/boost_testing_sidemade_demo.md:200]:
```suggestion
add_test(NAME "ConfigurationTests" COMMAND ${CMAKE_CURRENT_BINARY_DIR}/testsidemade --run_test=ConfigurationTests/*)
```
Closing parenthesis missing

**Review by ajaust — COMMENTED** [review]
Looks good overall. I only have very minor things like typos.

File: 05_testing_and_ci/boost_testing_intro_slides.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+    color: orange;
+  }
+  .reveal section img {
+    background:none;
+    border:none;
+    box-shadow:none;
+  }
+</style>
+
+# Introduction to Boost.Test
```
[the remark this answers is not in the document]
Comment (uekerman) [reply 05_testing_and_ci/boost_testing_intro_slides.md:35]: Learning goals were meant for the complete lecture, yes. This is sth we have to improve next year.

File: 05_testing_and_ci/boost_testing_intro_slides.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+See demo notes
+
+---
+
+## Test Suites
+
+- Organize tests as a tree
+    - Master test suite is root (handled internally, name specified with `BOOST_TEST_MODULE`)
+    - Test suites are internal nodes of the tree
+    - Test cases are the leaves
```
[the remark this answers is not in the document]
Comment (uekerman) [reply 05_testing_and_ci/boost_testing_intro_slides.md:83]: No, not a binary tree. How would you organize tests differently? In the end, it's just like a folder structure.

File: 05_testing_and_ci/boost_testing_sidemade_demo.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    - The code has no unit tests yet. -> We will some of them with Boost UTF.
+    - exercise: continue with same code, add more tests, set up GitHub Actions, and more
+
+## (2) First Dummy Test Case
+
+- `mkdir tests`, create `tests/MatrixSolverTest.cpp`
+- `#define BOOST_TEST_MODULE SideMadeTests`
+    - needs to be in one of our test files, we will write multiple ones
+- `#define BOOST_TEST_DYN_LINK`
+    - needed if UTF uses as dynamic library
```
[the remark this answers is not in the document]
Comment (uekerman) [reply 05_testing_and_ci/boost_testing_sidemade_demo.md:27]: With header-only, you cannot split tests over multiple files. Everything has to be in one cpp then.
Dynamic = shared here (is there actually a difference?). I will change to wording.

File: 05_testing_and_ci/boost_testing_intro_slides.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ajaust) [reply 05_testing_and_ci/boost_testing_intro_slides.md:83]: I did not expect anything special. As the data structure is mentioned it sounded to me as if it was something special/is important to the test framework. Therefore I asked. 

I don't see the problem to organize it as a graph, a list or whatever. Depends a bit on what operations the test framework has to do internally, but I am not an expert in this (=I have no idea what the framework has to do).

**Pull request merged**